File: resources/views/scan-feature/recognize.py
import sys
import requests
from io import BytesIO
from PIL import Image

# Brickognize synchronous recognition script
# Usage: python recognize.py <path/to-image> [parts|minifig|set]
# Default category is 'parts'

BASE_URL = "https://api.brickognize.com/predict"

def recognize_image(image_path, category="parts"):
    """
    Laadt de afbeelding in, zet 'm om naar PNG in-memory,
    en uploadt als 'query_image' naar het juiste predict-endpoint.
    """
    url = f"{BASE_URL}/{category}"

    # 1) Open en converteer naar PNG
    img = Image.open(image_path).convert("RGBA")
    buf = BytesIO()
    img.save(buf, format="PNG")
    buf.seek(0)

    # 2) Zet in multipart-form met expliciete PNG-mimetype
    files = {
        "query_image": (
            "upload.png",
            buf,
            "image/png"
        )
    }

    # 3) POST-en en eventueel debug-output tonen
    resp = requests.post(url, files=files, timeout=30)
    if resp.status_code != 200:
        print(">>> Response content:", resp.text)
    resp.raise_for_status()
    return resp.json()

def main():
    if len(sys.argv) < 2 or len(sys.argv) > 3:
        print("Usage: python recognize.py <image_path> [parts|minifig|set]")
        sys.exit(1)

    image_path = sys.argv[1]
    category   = sys.argv[2] if len(sys.argv) == 3 else "parts"

    try:
        result = recognize_image(image_path, category)
        print("Recognition result:")
        print(result)
    except Exception as e:
        print(f"Error during recognition: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: resources/views/scan-feature/recognize.js
// recognize.js
import fetch from "node-fetch";
import FormData from "form-data";
import fs from "fs";
import path from "path";

async function main() {
  const img = process.argv[2];
  if (!img || !fs.existsSync(img) || !fs.statSync(img).isFile()) {
    console.error("❌ Gebruik: node recognize.js <pad/naar/afbeelding>");
    process.exit(1);
  }

  const url = "https://api.brickognize.com/predict";
  const form = new FormData();
  form.append("query_image", fs.createReadStream(img), path.basename(img));

  console.log("📤 Uploaden:", img);
  const res = await fetch(url, { method: "POST", body: form });
  console.log("📡 HTTP status:", res.status);

  if (!res.ok) {
    console.error("🚨 Fout bij herkenning:", await res.text());
    process.exit(1);
  }

  const json = await res.json();
  console.log("✅ Herkenningsresultaat:\n", JSON.stringify(json, null, 2));
}

if (import.meta.url === `file://${process.argv[1]}`) main();
